src/logic/map_objects/tribes/soldiercontrol.cc

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2023 by the Widelands Development Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "logic/map_objects/tribes/soldiercontrol.h"
+
+#include <memory>
+
+#include "base/i18n.h"
+#include "base/string.h"
+#include "graphic/style_manager.h"
+#include "logic/map_objects/tribes/tribe_descr.h"
+
+namespace {
+
+std::string format_extra_soldiers(int has, int wants) {
+	assert(has > wants);
+	/** TRANSLATORS: %1% is the number of all soldiers in a warehouse, %2% is the desired
+	                 number */
+	return format(pgettext("warehouse", "%1% (%2%)"),
+	              StyleManager::color_tag(
+	                 as_string(has), g_style_manager->building_statistics_style().high_color()),
+	              wants);
+}
+
+}  // namespace
+
+namespace Widelands {
+
+#define CAPIDX TribeDescr::CapacityStringIndex
```

Comment:
Please don't do that, just spell it out below

Reply:
Sigh… I knew… But the long name makes it unreadable, and the confusing inconsistent line wrapping and indenting rules in our clang-format config will make it look even worse. Can't I use `using` or `namespace` somehow or something like that?